src/components/modals/TransactionAccelerateTransactionModal.js

```diff
@@ -0,0 +1,243 @@
+  guiWallet: GuiWallet,
+  walletDefaultDenomProps: EdgeDenomination
+}
+type StateProps = {
+  wallet: EdgeCurrencyWallet
+}
+
+type State = {
+  edgeUnsignedTransaction: ?EdgeTransaction,
+  error: ?Error,
+  status: Status
+}
+
+type Status = 'confirming' | 'sending' | 'sent'
+
+class TransactionAccelerateModalComponent extends PureComponent<Props, State> {
+  constructor() {
+    super()
+
+    this.state = {
+      edgeUnsignedTransaction: null,
+      error: null,
+      status: 'confirming'
+    }
+  }
+
+  componentDidMount() {
+    this.makeRbfTransaction()
+  }
+
+  componentWillUnmount() {
+    this.resetRbfState()
+  }
+
+  makeRbfTransaction = async () => {
+    const { edgeTransaction, wallet } = this.props
+
+    if (edgeTransaction.spendTargets) {
+      const spendTargets = edgeTransaction.spendTargets.map(spendTarget => ({
+        nativeAmount: spendTarget.nativeAmount,
+        publicAddress: spendTarget.publicAddress,
+        uniqueIdentifier: spendTarget.uniqueIdentifier
+      }))
+      const rbfTxid = edgeTransaction.txid
+
+      const edgeSpendInfo: EdgeSpendInfo = {
+        spendTargets,
+        rbfTxid
+      }
+
+      try {
+        const edgeUnsignedTransaction = await wallet.makeSpend(edgeSpendInfo)
+
+        this.setState({
+          edgeUnsignedTransaction
+        })
+      } catch (error) {
+        this.setState({
+          error: error
+        })
+      }
+    }
+  }
+
+  signBroadcastAndSaveRbf = async () => {
+    const { wallet } = this.props
+    const { edgeUnsignedTransaction } = this.state
+
+    if (edgeUnsignedTransaction) {
+      let edgeSignedTransaction: EdgeTransaction = edgeUnsignedTransaction
+
+      this.setState({ status: 'sending' })
+
+      try {
+        edgeSignedTransaction = await wallet.signTx(edgeUnsignedTransaction)
+        edgeSignedTransaction = await wallet.broadcastTx(edgeSignedTransaction)
+        await wallet.saveTx(edgeSignedTransaction)
+
+        // Should we save tx metadata? Where do we get the metadata?
+        // await wallet.saveTxMetadata(edgeSignedTransaction.txid, edgeSignedTransaction.currencyCode, edgeMetadata)
+
+        playSendSound().catch(error => console.log(error)) // Fail quietly
+
+        this.setState({ status: 'sent' })
+
+        Alert.alert(s.strings.transaction_success, s.strings.transaction_success_message, [
+          {
+            onPress() {},
+            style: 'default',
+            text: s.strings.string_ok
+          }
+        ])
+
+        Actions.replace(TRANSACTION_DETAILS, { edgeTransaction: edgeSignedTransaction })
+      } catch (error) {
+        console.log(error)
+        this.setState({ status: 'confirming' })
+        let message = sprintf(s.strings.transaction_failure_message, error.message)
+
+        Alert.alert(s.strings.transaction_failure, message, [
+          {
+            onPress() {},
+            style: 'default',
+            text: s.strings.string_ok
+          }
+        ])
```

Comment:
When unmounted: Don't set state.
Use showError Airship instance instead of `Alert.alert`.